tools/karpenter-convert/pkg/convert/convert.go

```diff
@@ -161,20 +190,30 @@ func processNodeTemplate(resource runtime.Object) runtime.Object {
 	}
 
 	nodeclass := nodeclassutil.New(nodetemplate)
+	nodeclass.ObjectMeta.CreationTimestamp = metav1.Time{}
 	nodeclass.TypeMeta = metav1.TypeMeta{
 		Kind:       "EC2NodeClass",
 		APIVersion: v1beta1.SchemeGroupVersion.String(),
 	}
-	nodeclass.Spec.Role = "<your AWS role here>"
+	nodeclass.Spec.Role = "$KARPENTER_NODE_ROLE"
 	return nodeclass
 }
 
-func processProvisioner(resource runtime.Object) runtime.Object {
-	provisioner := resource.(*v1alpha5.Provisioner)
-	nodepool := nodepoolutil.New(provisioner)
+func processProvisioner(resource runtime.Object, o *Context) runtime.Object {
+	coreprovisioner := resource.(*corev1alpha5.Provisioner)
+
+	if !o.IgnoreDefaults {
+		provisioner := lo.ToPtr(v1alpha5.Provisioner(lo.FromPtr(coreprovisioner)))
+		provisioner.SetDefaults(context.Background())
+		coreprovisioner = lo.ToPtr(corev1alpha5.Provisioner(lo.FromPtr(provisioner)))
+	}
+
+	nodepool := nodepoolutil.New(coreprovisioner)
+	nodepool.ObjectMeta.CreationTimestamp = metav1.Time{}
```

Comment:
```suggestion
	nodepool.CreationTimestamp = metav1.Time{}
```

nit: Can remove the `ObjectMeta` portion